cargo-dist/templates/installer/homebrew.rb.j2

```diff
@@ -1,70 +1,32 @@
 class {{ formula_class }} < Formula
-  {%- if desc %}
   desc "{{ desc }}"
-  {%- endif %}
-  {%- if homepage %}
+  {%- if homepage is defined %}
   homepage "{{ homepage }}"
   {%- endif %}
   version "{{ inner.app_version }}"
-  {#- #}
-  {%- if arm64_macos.id or x86_64_macos.id %}
-  if OS.mac?
-    {#- If arm64/x86_64 builds are the same, skip the Hardware::CPU.type if statement #}
-    {%- if arm64_macos.id == x86_64_macos.id %}
-    url "{{ inner.base_url }}/{{ arm64_macos.id }}"
-    {%- if arm64_macos_sha256 %}
-    sha256 "{{ arm64_macos_sha256 }}"
-    {%- endif %}
-    {%- else %}
-    {%- if arm64_macos.id %}
-    if Hardware::CPU.arm?
-      url "{{ inner.base_url }}/{{ arm64_macos.id }}"
-      {%- if arm64_macos_sha256 %}
-      sha256 "{{ arm64_macos_sha256 }}"
-      {%- endif %}
-    end
-    {%- endif %}
-    {%- if x86_64_macos.id %}
-    if Hardware::CPU.intel?
-      url "{{ inner.base_url }}/{{ x86_64_macos.id }}"
-      {%- if x86_64_macos_sha256 %}
-      sha256 "{{ x86_64_macos_sha256 }}"
-      {%- endif %}
-    end
-    {%- endif %}
-    {%- endif %}
-  end
-  {%- endif %}
-  {#- #}
-  {%- if arm64_linux.id or x86_64_linux.id %}
-  if OS.linux?
-    {%- if arm64_linux.id %}
-    if Hardware::CPU.arm?
-      url "{{ inner.base_url }}/{{ arm64_linux.id }}"
-      {%- if arm64_linux %}
-      sha256 "{{ arm64_linux_sha256 }}"
-      {%- endif %}
-    end
-    {%- endif %}
-    {%- if x86_64_linux.id %}
-    if Hardware::CPU.intel?
-      url "{{ inner.base_url }}/{{ x86_64_linux.id }}"
-      {%- if x86_64_linux_sha256 %}
-      sha256 "{{ x86_64_linux_sha256 }}"
-      {%- endif %}
-    end
+
+  {% macro fragment_url_and_hash(cond, frag) -%}
```

Comment:
A comment to note this is a jinja macro could be helpful - we don't use a lot of jinja macros (and the `macro` keyword is a little overloaded given we're writing Rust). It'll help make it clearer to the next person looking at this.